src/org/labkey/test/TestSet.java

```diff
@@ -90,6 +92,40 @@ public List<Class<?>> getTestList()
         return _tests;
     }
 
+    public List<Class<?>> getSortedTestList()
+    {
+        List<Class<?>> firstTests = new ArrayList<>();
+        List<Class<?>> unprioritized = new ArrayList<>();
+        List<Class<?>> lastTests = new ArrayList<>();
+        for (Class<?> next : _tests)
+        {
+            Prioritized annotation = next.getAnnotation(Prioritized.class);
+            double priority = annotation != null ? annotation.priority() : 0;
+            if (priority > 0)
+            {
+                firstTests.add(next);
+            }
+            else if (priority < 0)
+            {
+                lastTests.add(next);
+            }
+            else
+            {
+                unprioritized.add(next);
+            }
+        }
+        firstTests.sort(Comparator.comparingDouble(c -> c.getAnnotation(Prioritized.class).priority()));
+        Collections.reverse(firstTests);
+        lastTests.sort(Comparator.comparingDouble(c -> c.getAnnotation(Prioritized.class).priority()));
+        Collections.reverse(lastTests);
```

Comment:
I think I understand but want to make sure. The `Collections.reverse` is used to make sure that a test with priority 2, runs before 1 because 2 is a higher priority. Is that correct? 

If that is correct, do you want the reverse on the `lastTests`? Test in that list have a priority that is negative, and in that case wouldn't you want -1 to run before -2?

Reply:
Yeah. The default comparator sorts [-2, -1], which is the reverse of what we want.